### ISSUE TITLE
Fast Refresh does not work @latest

### ISSUE THREAD

**Issue by kasperaamodt** [vercel/next.js#35703]
### Verify canary release

- [X] I verified that the issue exists in Next.js canary release

### Provide environment information

    Operating System:
      Platform: darwin
      Arch: arm64
      Version: Darwin Kernel Version 21.3.0: Wed Jan  5 21:37:58 PST 2022; root:xnu-8019.80.24~20/RELEASE_ARM64_T8101
    Binaries:
      Node: 17.5.0
      npm: 8.4.1
      Yarn: N/A
      pnpm: N/A
    Relevant packages:
      next: 12.1.2
      react: 18.0.0
      react-dom: 18.0.0

warn  - Latest canary version not detected, detected: "12.1.2", newest: "12.1.3-canary.0".
        Please try the latest canary version (`npm install next@canary`) to confirm the issue still exists before creating a new issue.
        Read more - https://nextjs.org/docs/messages/opening-an-issue

### What browser are you using? (if relevant)

Chrome 99.0.4844.83 (Official Build) (arm64)

### Describe the Bug

Just installed a new project with `npx create-next-app@latest`, and fast refresh is not working. 

### Expected Behavior

For the browser to refresh when I save a change in my code editor.

### To Reproduce

`npx create-next-app@latest`

start project with `npm run dev`

**Comment by kasperaamodt** [comment]
Updated to Chrome Version 100.0.4896.60 (Official Build) (arm64), still same issue.

**Comment by kasperaamodt** [comment]
Correction on my ticket. It appears the fast refresh is working, it's just taking a long time for it to actually reload. I measured it to about 4 seconds.

**Comment by lordkerwin** [comment]
I have the same issue, kind of.
Brand new nextjs app, with/without typescript, created using `npx create-next-app@latest` and i can see in the terminal that it's reloading.

```
event - compiled client and server successfully in 107 ms (142 modules)
```

and in the browser (chrome, firefox and brave) I see the little vercel loader in the bottom right for a fraction of a second... however, the page doesn't refresh.

Like i said, brand new next app, and all i have changed is the word `Welcome` to `Hello` and i have to actually go and `F5` or refresh the page to see changes.

Forgot to mention, latest LTS node, running on Pop!_OS.

**Comment by kasperaamodt** [comment]
I found out that I had a component that made my website refresh every 4 seconds, so when I removed that component, the fast refresh does not work at all.

**Comment by CarlosLevir** [comment]
Same thing here.

**Comment by bryanbarrios** [comment]
Same problem here.  I'm using the @latest version on an M1.

**Comment by djxy** [comment]
For those who have this issue. You need to downgrade the react version to 17.0.2 for `react` and `react-dom`.

**Comment by caestrada** [comment]
@djxy that was it!

**Comment by diegohaz** [comment]
Fast Refresh also stopped working for me. The page only gets updated when I interact with an element with an internal state (and the state changes).

**Comment by timneutkens** [comment]
Figured out the problem and opened a PR here: https://github.com/vercel/next.js/pull/35718

**Comment by timneutkens** [comment]
Released in [12.1.3](https://github.com/vercel/next.js/releases/tag/v12.1.3)
